core/src/test/java/io/grpc/ClientInterceptorsTest.java

```diff
@@ -351,6 +352,33 @@ protected void checkedStart(ClientCall.Listener<RespT> responseListener, Metadat
     ArgumentCaptor<Status> captor = ArgumentCaptor.forClass(Status.class);
     verify(listener).onClose(captor.capture(), any(Metadata.class));
     assertSame(error, captor.getValue().getCause());
+
+    // Make sure nothing bad happens after the exception.
+    ClientCall<?, ?> noop = ((CheckedForwardingClientCall<?, ?>)interceptedCall).delegate();
```

Comment:
verifyNoMoreInteractions is the path to tight coupling.  I avoid it whenever possible.  Mocking would be much less painful if it was used sparingly.  The benefit it adds isn't worth the maintenance burden.  

The code above probably isn't doing what you think it is, because it didn't cover the lines that this addition is.